src/app/EventManagement.cpp

```diff
@@ -748,6 +751,71 @@ CHIP_ERROR EventManagement::FetchEventsSince(TLVWriter & aWriter, const ObjectLi
     return err;
 }
 
+CHIP_ERROR EventManagement::RemoveInvalidFabricCB(const TLV::TLVReader & aReader, size_t aDepth, void * apContext)
+{
+    // the function does not actually remove the event, instead, it sets the fabric index to an invalid value.
+    FabricIndex * invalidFabricIndex = static_cast<FabricIndex *>(apContext);
+
+    TLVReader event;
+    TLVType tlvType;
+    TLVType tlvType1;
+    event.Init(aReader);
+    ReturnErrorOnFailure(event.EnterContainer(tlvType));
+    ReturnErrorOnFailure(event.Next());
+    ReturnErrorOnFailure(event.EnterContainer(tlvType1));
+    ReturnErrorOnFailure(event.Next());
+
+    uint8_t fabricIndex = 0;
+    while (CHIP_NO_ERROR == event.Next())
+    {
+        if (event.GetTag() == TLV::ProfileTag(kEventManagementProfile, kFabricIndexTag))
+        {
+            ReturnErrorOnFailure(event.Get(fabricIndex));
+            if (fabricIndex == *invalidFabricIndex)
+            {
+                // fabricIndex is encoded as an integer; the dataPtr will point to a location immediately after its encoding
+                uint8_t * dataPtr                  = const_cast<uint8_t *>(event.GetReadPoint());
```

Comment:
Would be good to avoid the const_cast here.  Specifically, maybe we should just use `dataPtr` to compute an offset from `readBuffer->GetQueue()` and then add that offset to the already non-const `GetQueue()` value?

Reply:
Done, thanks